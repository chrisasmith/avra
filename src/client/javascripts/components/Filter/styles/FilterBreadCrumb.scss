@import '../../../../stylesheets/modules/theme.scss';

.breadcrumb {
  padding-left: 16px;
  font-size: 15px;
}

.label {
  width: 125px;
  font-weight: bold;
  display: inline-block;
}

