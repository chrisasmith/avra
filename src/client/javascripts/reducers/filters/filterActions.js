import { api } from '../../services'
import { createRequests, doApiCall, createRequestTypes } from '../actionHelper'

// test data
export const DEFAULT_FILTER = {
  property: { name: 'Premier League', id: 32, isGroup: false, type: 'league', sport: 'Football / Soccer' },
  benchmark: { name: 'All Sports Fans', id: 0, isGroup: false },
  country: { name: 'All Countries', id: 1, isGroup: true },
  ages: {
    men: [0, 1, 2],
    women: [0, 1, 2],
  },
  income: [0, 1, 2],
  kids: [0, 1],
  interest: [0, 1, 2, 3, 4, 5],
}

export const DEFAULT_OPTIONS = {
  filter: [{ id: 0, name: 'hey there' }],
  benchmark: [
    { name: 'All Sports Fans', id: 0, isGroup: false },
  ],
  country: [
    { name: 'All Countries', id: 1, isGroup: true },
    { name: 'Finland', id: 3, isGroup: false },
    { name: 'Canada', id: 4, isGroup: false },
    {
      name: 'United States',
      id: 5,
      isGroup: false,
      children: [
        { name: 'All USA', id: 4, isGroup: true },
        { name: 'Atlanta', id: 5, isGroup: false },
        { name: 'Chicago', id: 6, isGroup: false },
        { name: 'Seattle', id: 7, isGroup: false },
      ],
    },
  ],
  property: [
    { name: 'American Football', id: 31, isGroup: false, type: 'sport' },
    { name: 'Athletics / Track and Field', id: 1, isGroup: false, type: 'sport' },
    { name: 'Badminton', id: 2, isGroup: false, type: 'sport' },
    { name: 'Baseball', id: 4, isGroup: false, type: 'sport' },
    {
      name: 'Basketball',
      id: 3,
      isGroup: false,
      children: [
        { name: 'All Basketball', id: 3, isGroup: true, type: 'sport', sport: 'Basketball' },
        { name: 'IBL', id: 4, isGroup: false, type: 'league', sport: 'Basketball' },
        {
          name: 'NBA',
          id: 5,
          isGroup: false,
          children: [
            { name: 'All NBA', id: 5, isGroup: true, type: 'league', sport: 'Basketball' },
            { name: 'Atlanta Hawks', id: 6, isGroup: false, type: 'team', sport: 'Basketball' },
            { name: 'Boston Celtics', id: 7, isGroup: false, type: 'team', sport: 'Basketball' },
            { name: 'Brooklyn Nets', id: 8, isGroup: false, type: 'team', sport: 'Basketball' },
          ],
        },
      ],
    },
    { name: 'Boxing', id: 5, isGroup: false, type: 'sport' },
    { name: 'Cricket', id: 6, isGroup: false, type: 'sport' },
    { name: 'Cycling', id: 7, isGroup: false, type: 'sport' },
    { name: 'Darts', id: 8, isGroup: false, type: 'sport' },
    { name: 'Endurance Running', id: 9, isGroup: false, type: 'sport' },
    { name: 'Extreme Sports / Action sports', id: 10, isGroup: false, type: 'sport' },
    { name: 'Fencing', id: 11, isGroup: false, type: 'sport' },
    { name: 'Field Hockey', id: 12, isGroup: false, type: 'sport' },
    {
      name: 'Football / Soccer',
      id: 13,
      isGroup: false,
      type: 'sport',
      children: [
        { name: 'All Football / Soccer', id: 3, isGroup: true, type: 'sport', sport: 'Football / Soccer' },
        { name: 'Premier League', id: 32, isGroup: false, type: 'league', sport: 'Football / Soccer' },
      ],
    },
    { name: 'Golf', id: 14, isGroup: false, type: 'sport' },
    { name: 'Handball', id: 15, isGroup: false, type: 'sport' },
    { name: 'Ice Hockey', id: 16, isGroup: false, type: 'sport' },
    { name: 'Mixed Martial Arts (MMA)', id: 17, isGroup: false, type: 'sport' },
    { name: 'Motorsport', id: 18, isGroup: false, type: 'sport' },
    { name: 'Netball', id: 19, isGroup: false, type: 'sport' },
    { name: 'Rugby League', id: 20, isGroup: false, type: 'sport' },
    { name: 'Rugby Union / Rugby', id: 21, isGroup: false, type: 'sport' },
    { name: 'Sailing', id: 22, isGroup: false, type: 'sport' },
    { name: 'Snooker', id: 23, isGroup: false, type: 'sport' },
    { name: 'Surfing', id: 24, isGroup: false, type: 'sport' },
    { name: 'Swimming', id: 25, isGroup: false, type: 'sport' },
    { name: 'Tennis', id: 27, isGroup: false, type: 'sport' },
    { name: 'Triathlon', id: 28, isGroup: false, type: 'sport' },
    { name: 'Volleyball', id: 29, isGroup: false, type: 'sport' },
    { name: 'Winter Sports', id: 30, isGroup: false, type: 'sport' },
  ],
}

export const UPDATE_FILTER = 'UPDATE_FILTER'
export function updateFilter(filter) {
  return { type: UPDATE_FILTER, filter }
}

export const SAVE_FILTER = createRequestTypes('SAVE_FILTER')
export const saveFilterAction = createRequests(SAVE_FILTER)

export function saveFilter(filterName, filter) {
  return doApiCall({
    api: api.saveFilter,
    apiParams: { filterName, filter },
    action: saveFilterAction,
  })
}
