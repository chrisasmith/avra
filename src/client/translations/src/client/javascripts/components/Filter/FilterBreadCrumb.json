[
  {
    "id": "bread_crumb.segmented",
    "defaultMessage": "Segmented by:"
  },
  {
    "id": "bread_crumb.benchmark",
    "defaultMessage": "Benchmark:"
  }
]