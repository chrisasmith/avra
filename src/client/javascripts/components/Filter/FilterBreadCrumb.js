import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import styles from './styles/FilterBreadCrumb.scss'
import { DEFAULT_FILTER } from '../../reducers/filters/filterActions'

const propTypes = {
  filter: PropTypes.object.isRequired,
}

const compareArray = (arr1, arr2) => arr1.length === arr2.length && arr1.sort().every((value, index) => value === arr2.sort()[index])
const addComma = str => str.length > 0 ? ', ' : ''

const FilterBreadCrumb = ({ filter }) => {
  let filterString = ''

  if (filter.country.id !== DEFAULT_FILTER.country.id) {
    filterString = 'Country'
  }

  if (!compareArray(filter.ages.men, DEFAULT_FILTER.ages.men) || !compareArray(filter.ages.women, DEFAULT_FILTER.ages.women)) {
    filterString += `${addComma(filterString)}Age & gender`
  }

  if (!compareArray(filter.income, DEFAULT_FILTER.income)) {
    filterString += `${addComma(filterString)}Income`
  }

  if (!compareArray(filter.kids, DEFAULT_FILTER.kids)) {
    filterString += `${addComma(filterString)}Kids in household`
  }

  if (!compareArray(filter.interest, DEFAULT_FILTER.interest)) {
    filterString += `${addComma(filterString)}Interest level`
  }

  return (
    <div className={styles.breadcrumb}>
      <div>
        <span className={styles.label}>
          <FormattedMessage
            id="bread_crumb.segmented"
            defaultMessage="Segmented by:"
          />
        </span>
        <span>{filterString.length > 0 ? filterString : 'None'}</span>
      </div>
      <div>
        <span className={styles.label}>
          <FormattedMessage
            id="bread_crumb.benchmark"
            defaultMessage="Benchmark:"
          />
        </span>
        <span>{filter.benchmark.name}</span>
      </div>
    </div>
  )
}

FilterBreadCrumb.propTypes = propTypes
export default FilterBreadCrumb
