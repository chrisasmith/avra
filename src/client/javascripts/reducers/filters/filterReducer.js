import * as ActionTypes from './filterActions'

export default function reducer(state = {
  filter: { ...ActionTypes.DEFAULT_FILTER, ages: { ...ActionTypes.DEFAULT_FILTER.ages } },
  options: { ...ActionTypes.DEFAULT_OPTIONS },
}, action) {
  switch (action.type) {
    case ActionTypes.UPDATE_FILTER: {
      return { ...state, filter: action.filter }
    }

    default:
      return state
  }
}
