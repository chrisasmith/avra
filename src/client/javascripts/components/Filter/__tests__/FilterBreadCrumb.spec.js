import React from 'react'
import { mount } from 'enzyme'
import { renderToJson } from 'enzyme-to-json'
import FilterBreadCrumb from '../FilterBreadCrumb'
import { DEFAULT_FILTER } from '../../../reducers/filters/filterActions'

describe('Components FilterBreadCrumb', () => {
  it('should render', () => {
    const wrapper = mount(<FilterBreadCrumb
      filter={DEFAULT_FILTER}
    />)

    expect(renderToJson(wrapper.render())).toMatchSnapshot()
  })

  it('should render country', () => {
    const wrapper = mount(<FilterBreadCrumb
      filter={{ ...DEFAULT_FILTER, country: { id: 5 } }}
    />)

    expect(renderToJson(wrapper.render())).toMatchSnapshot()
  })

  it('should render age', () => {
    const wrapper = mount(<FilterBreadCrumb
      filter={{ ...DEFAULT_FILTER, ages: { men: [1], women: [0] } }}
    />)

    expect(renderToJson(wrapper.render())).toMatchSnapshot()
  })

  it('should render income', () => {
    const wrapper = mount(<FilterBreadCrumb
      filter={{ ...DEFAULT_FILTER, income: [0] }}
    />)

    expect(renderToJson(wrapper.render())).toMatchSnapshot()
  })

  it('should render kids', () => {
    const wrapper = mount(<FilterBreadCrumb
      filter={{ ...DEFAULT_FILTER, kids: [0] }}
    />)

    expect(renderToJson(wrapper.render())).toMatchSnapshot()
  })

  it('should render interest', () => {
    const wrapper = mount(<FilterBreadCrumb
      filter={{ ...DEFAULT_FILTER, interest: [0] }}
    />)

    expect(renderToJson(wrapper.render())).toMatchSnapshot()
  })

  it('should render all of them', () => {
    const wrapper = mount(<FilterBreadCrumb
      filter={{ ...DEFAULT_FILTER, country: { id: 5 }, ages: { men: [1], women: [0] }, income: [0], kids: [0], interest: [0] }}
    />)

    expect(renderToJson(wrapper.render())).toMatchSnapshot()
  })
})
