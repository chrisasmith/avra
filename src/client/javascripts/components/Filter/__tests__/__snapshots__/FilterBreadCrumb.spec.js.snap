// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components FilterBreadCrumb should render 1`] = `
<div
  class="breadcrumb"
>
  <div>
    <span
      class="label"
    >
      Segmented by:
    </span>
    <span>
      None
    </span>
  </div>
  <div>
    <span
      class="label"
    >
      Benchmark:
    </span>
    <span>
      All Sports Fans
    </span>
  </div>
</div>
`;

exports[`Components FilterBreadCrumb should render age 1`] = `
<div
  class="breadcrumb"
>
  <div>
    <span
      class="label"
    >
      Segmented by:
    </span>
    <span>
      Age & gender
    </span>
  </div>
  <div>
    <span
      class="label"
    >
      Benchmark:
    </span>
    <span>
      All Sports Fans
    </span>
  </div>
</div>
`;

exports[`Components FilterBreadCrumb should render all of them 1`] = `
<div
  class="breadcrumb"
>
  <div>
    <span
      class="label"
    >
      Segmented by:
    </span>
    <span>
      Country, Age & gender, Income, Kids in household, Interest level
    </span>
  </div>
  <div>
    <span
      class="label"
    >
      Benchmark:
    </span>
    <span>
      All Sports Fans
    </span>
  </div>
</div>
`;

exports[`Components FilterBreadCrumb should render country 1`] = `
<div
  class="breadcrumb"
>
  <div>
    <span
      class="label"
    >
      Segmented by:
    </span>
    <span>
      Country
    </span>
  </div>
  <div>
    <span
      class="label"
    >
      Benchmark:
    </span>
    <span>
      All Sports Fans
    </span>
  </div>
</div>
`;

exports[`Components FilterBreadCrumb should render income 1`] = `
<div
  class="breadcrumb"
>
  <div>
    <span
      class="label"
    >
      Segmented by:
    </span>
    <span>
      Income
    </span>
  </div>
  <div>
    <span
      class="label"
    >
      Benchmark:
    </span>
    <span>
      All Sports Fans
    </span>
  </div>
</div>
`;

exports[`Components FilterBreadCrumb should render interest 1`] = `
<div
  class="breadcrumb"
>
  <div>
    <span
      class="label"
    >
      Segmented by:
    </span>
    <span>
      Interest level
    </span>
  </div>
  <div>
    <span
      class="label"
    >
      Benchmark:
    </span>
    <span>
      All Sports Fans
    </span>
  </div>
</div>
`;

exports[`Components FilterBreadCrumb should render kids 1`] = `
<div
  class="breadcrumb"
>
  <div>
    <span
      class="label"
    >
      Segmented by:
    </span>
    <span>
      Kids in household
    </span>
  </div>
  <div>
    <span
      class="label"
    >
      Benchmark:
    </span>
    <span>
      All Sports Fans
    </span>
  </div>
</div>
`;
